lms/djangoapps/discussion/templates/discussion/discussion_mfe_embed_js.template

```diff
@@ -0,0 +1,35 @@
+## mako
+
+<%!
+
+from openedx.core.djangolib.js_utils import dump_js_escaped_json, js_escaped_string
+from cms.djangoapps.contentstore import utils
+%>
+
+const domain = (new URL("${discussions_mfe_url}"));
+const course_id = "${course_key}"; 
+let current_pathname = window.location.pathname.slice();
+window.addEventListener("message", (event) => {
+    if ((event.origin !== domain.origin) || (event.data.type !== "discussions-mfe-tab-embed"))
+        return;
+    const {postId ,topicId} = event.data.payload;
+    if (topicId){
+        let new_url = "".concat(
+            current_pathname.split('discussion/forum/')[0],
+            "discussion/forum/",
+            topicId
+            )
+        if (postId){
+            new_url = new_url.concat("/threads/",postId);
+        }
+        window.history.pushState(null,null , new_url);
+    }
+    else{
+        let new_url = "".concat(
+            current_pathname.split('discussion/forum/')[0],
+            "discussion/forum/",
+            )
+        window.history.pushState(null,null , new_url);
+    }
```

Comment:
I think this code can be greatly simplified. 
```suggestion
    const url = current_pathname.split('discussion/forum/')[0].concat("discussion/forum/");
    if (topicId) {
        url += topicId;
        if (postId){
            url = url.concat("/threads/",postId);
        }
    }
    window.history.pushState(null, null, url);
```

Also, instead of getting doing `current_pathname.split` you can just pass in the root URL from the LMS. Look at how the root_url is being passed now and used.

Reply:
Yes I am starting using `mfe_url`